fga/store/cachedFgaTenantStore.go

```diff
@@ -0,0 +1,96 @@
+package store
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"slices"
+	"time"
+
+	"github.com/hashicorp/golang-lru/v2/expirable"
+	openfgav1 "github.com/openfga/api/proto/openfga/v1"
+	"google.golang.org/grpc/status"
+)
+
+type FgaTenantStore interface {
+	GetStoreIDForTenant(ctx context.Context, conn openfgav1.OpenFGAServiceClient, tenantID string) (string, error)
+	GetModelIDForTenant(ctx context.Context, conn openfgav1.OpenFGAServiceClient, tenantID string) (string, error)
+	IsDuplicateWriteError(err error) bool
+	GetCache() *expirable.LRU[string, string]
```

Comment:
I dont think we should expose the cache to the consumer of the library

Reply:
done